api-server/scanner-lib/src/sync/tests/mod.rs

```diff
@@ -1257,3 +1261,663 @@ async fn check_all_destinations_are_tracked(#[case] seed: Seed) {
+
+    let target_block_time = chain_config.target_block_spacing();
+    let genesis_id = chain_config.genesis_block_id();
+
+    // Create Spend and Refund destinations
+    let (spend_sk, spend_pk) = PrivateKey::new_from_rng(&mut rng, KeyKind::Secp256k1Schnorr);
+    let spend_dest = Destination::PublicKey(spend_pk.clone());
+
+    let (refund_sk, refund_pk) = PrivateKey::new_from_rng(&mut rng, KeyKind::Secp256k1Schnorr);
+    let refund_dest = Destination::PublicKey(refund_pk.clone());
+
+    // Construct HTLC Data
+    let secret = HtlcSecret::new_from_rng(&mut rng);
+    let secret_hash = secret.hash();
+
+    let htlc_data = HashedTimelockContract {
+        secret_hash,
+        spend_key: spend_dest.clone(),
+        refund_key: refund_dest.clone(),
+        refund_timelock: OutputTimeLock::ForBlockCount(1),
+    };
+
+    // Create Transaction with 2 HTLC outputs
+    let tx_fund = TransactionBuilder::new()
+        .add_input(
+            TxInput::from_utxo(OutPointSourceId::BlockReward(genesis_id), 0),
+            InputWitness::NoSignature(None),
+        )
+        .add_output(TxOutput::Htlc(
+            OutputValue::Coin(Amount::from_atoms(1000)),
+            Box::new(htlc_data.clone()),
+        ))
+        .add_output(TxOutput::Htlc(
+            OutputValue::Coin(Amount::from_atoms(1000)),
+            Box::new(htlc_data),
+        ))
+        .build();
+
+    let tx_fund_id = tx_fund.transaction().get_id();
+
+    // Create and Process Block
+    tf.progress_time_seconds_since_epoch(target_block_time.as_secs());
+    let block = tf
+        .make_block_builder()
+        .with_parent(genesis_id)
+        .with_transactions(vec![tx_fund.clone()])
+        .build(&mut rng);
+
+    tf.process_block(block.clone(), BlockSource::Local).unwrap();
+    local_state.scan_blocks(BlockHeight::new(1), vec![block]).await.unwrap();
+
+    // Verify Storage
+    let db_tx = local_state.storage().transaction_ro().await.unwrap();
+
+    // Check Spend Address Transactions
+    let spend_address = Address::new(&chain_config, spend_dest).unwrap();
+    let spend_txs = db_tx.get_address_transactions(spend_address.as_str()).await.unwrap();
+    assert!(
+        spend_txs.contains(&tx_fund_id),
+        "Spend address should track the transaction"
+    );
+
+    // Check Refund Address Transactions
+    let refund_address = Address::new(&chain_config, refund_dest).unwrap();
+    let refund_txs = db_tx.get_address_transactions(refund_address.as_str()).await.unwrap();
+    assert!(
+        refund_txs.contains(&tx_fund_id),
+        "Refund address should track the transaction"
+    );
+
+    let utxos = db_tx.get_address_available_utxos(spend_address.as_str()).await.unwrap();
+    assert_eq!(utxos.len(), 2);
+    assert!(utxos.iter().map(|(outpoint, _)| outpoint).any(
+        |outpoint| *outpoint == UtxoOutPoint::new(OutPointSourceId::Transaction(tx_fund_id), 0)
+    ));
```

Comment:
1. You repeat these `UtxoOutPoint::new(OutPointSourceId::Transaction(tx_fund_id), x)` multiple times all over the test. Plz put them into variables.

2. Call to `map` is redundant, it just adds extra noise.

3. IMO it's still better to form a set of outpoints and directly compare it to the set of expected outpoints, instead of doing the length comparison followed by the assertions on `any`.